### DISCUSSION TITLE
Is sqlg affected by the vulnerability of Apache Commons Text?

### DISCUSSION THREAD

**Discussion by antheil** [pietermartin/sqlg#468] [Q&A] [answered]
A vulnerability was found in Apache Commons Text recently. Is slqg affected by that?
Details could be found in: https://seclists.org/oss-sec/2022/q4/22
CVE-2022-42889: Apache Commons Text prior to 1.10.0 allows RCE when applied to untrusted input due to insecure interpolation defaults
Thanks.
@pietermartin @diegonc @ssalevan @metlos @JPMoresmau

**Comment by pietermartin** [comment, marked as answer]
No, a search in Sqlg shows that Sqlg itself only uses,
`org.apache.commons.text.RandomStringGenerator`
`org.apache.commons.text.StringEscapeUtils`

If I read the vulnerability correctly it refers to `org.apache.commons.text.lookup.StringLookup`

Apache Commons Text is however also used by `TinkerPop's`  `gremlin-core` and `gremlin-language` and `gremlin-test`.
Searching in there is looks like it only uses,

`org.apache.commons.text.StringEscapeUtils`
`org.apache.commons.text.TextStringBuilder`
